rskj-core/src/main/java/co/rsk/peg/lockingcap/LockingCapStorageProviderImpl.java

```diff
@@ -0,0 +1,52 @@
+package co.rsk.peg.lockingcap;
+
+import static co.rsk.peg.lockingcap.LockingCapStorageIndexKey.LOCKING_CAP;
+import static org.ethereum.config.blockchain.upgrades.ConsensusRule.RSKIP134;
+
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.peg.BridgeSerializationUtils;
+import co.rsk.peg.storage.StorageAccessor;
+import java.util.Optional;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+
+public class LockingCapStorageProviderImpl implements LockingCapStorageProvider {
+
+    private Coin lockingCap;
+    private final StorageAccessor bridgeStorageAccessor;
+
+    public LockingCapStorageProviderImpl(StorageAccessor bridgeStorageAccessor) {
+        this.bridgeStorageAccessor = bridgeStorageAccessor;
+    }
+
+    @Override
+    public Optional<Coin> getLockingCap(ActivationConfig.ForBlock activations) {
+        if (activations.isActive(RSKIP134)) {
+            if (lockingCap == null) {
+                lockingCap = initializeLockingCap();
+            }
+            return Optional.of(lockingCap);
+        }
+        return Optional.empty();
+    }
+
+    private synchronized Coin initializeLockingCap() {
+        return bridgeStorageAccessor.getFromRepository(LOCKING_CAP.getKey(), BridgeSerializationUtils::deserializeCoin);
```

Comment:
```suggestion
    private synchronized void initializeLockingCap() {
lockingCap = bridgeStorageAccessor.getFromRepository(LOCKING_CAP.getKey(), BridgeSerializationUtils::deserializeCoin);
```
If you want to instantiate a class variable, then simply assign the value. No need to return I guess

Reply:
I agree with this approach

Reply:
Done!